src/Support/SupportService.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * Service class for support.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Support;
+
+use AmpProject\AmpWP\Infrastructure\Service;
+use AmpProject\AmpWP\Infrastructure\CliCommand;
+use AmpProject\AmpWP\Validation\ScannableURLProvider;
+use AmpProject\AmpWP\Validation\URLScanningContext;
+
+/**
+ * Class SupportService
+ * Service class for support.
+ */
+class SupportService implements Service, CliCommand {
+
+	/**
+	 * Get the name under which to register the CLI command.
+	 *
+	 * @return string The name under which to register the CLI command.
+	 */
+	public static function get_command_name() {
+
+		return 'amp';
+	}
+
+	/**
+	 * To send support data.
+	 *
+	 * @param array $args Support data argument.
+	 *
+	 * @return array|\WP_Error Response from insight server.
+	 */
+	public function send_data( $args = [] ) {
+
+		$support = new SupportData( $args );
+
+		return $support->send_data();
+	}
+
+	/**
+	 * To get support data.
+	 *
+	 * @param array $args Support data argument.
+	 *
+	 * @return array Support data.
+	 */
+	public function get_data( $args = [] ) {
+
+		$support = new SupportData( $args );
+
+		return $support->get_data();
+	}
+
+	/**
+	 * Sends support data to endpoint.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--is-synthetic]
+	 * : Whether or not it is synthetic data.
+	 * ---
+	 * default: false
+	 * options:
+	 *   - true
+	 *   - false
+	 *
+	 * [--print]
+	 * : To print support data.
+	 * ---
+	 * default: json-pretty
+	 * options:
+	 *   - json
+	 *   - json-pretty
+	 *
+	 * [--endpoint=<string>]
+	 * : Support endpoint. Where support data will send.
+	 *
+	 * [--urls=<urls>]
+	 * : List of URL for which support data need to send. Use comma separator for multiple URLs.
+	 *
+	 * [--post_ids=<post_ids>]
+	 * : List of Post for which support data need to send. Use comma separator for multiple post ids.
+	 *
+	 * [--term_ids=<term_ids>]
+	 * : List of term for which support data need to send. Use comma separator for multiple term ids.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp amp send-diagnostic
+	 *
+	 * @subcommand send-diagnostic
```

Comment:
In the same way as was done for validation, with `wp amp validation check-url`, there should probably be a `support` subcommand under `amp` like `validation`. And then the verb to send diagnostic can be a subcommand under that, so: `wp amp support send-diagnostic`.